### PR TITLE
[*] Custom components

### PR DESCRIPTION
**Pull request by arhtudormorar**

### CODE REVIEW

File: src/hooks/transactions/useCheckTransactionStatus/manageFailedTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}: {
+  results: SmartContractResult[];
+  hash: string;
+  sessionId: string;
+}) {
+  const resultWithError = results.find(
+    (scResult) => scResult?.returnMessage !== ''
+  );
+
+  store.dispatch(
```
Comment (StanislavSava) [inline src/hooks/transactions/useCheckTransactionStatus/manageFailedTransactions.ts:25]: Do we need 2 dispatches here? I feel like one is redundant

File: src/components/TransactionsTracker/useTransactionsTracker.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const pendingBatches = pendingTransactionsArray.filter(
+    ([sessionId, transactionBatch]) => {
+      const isPending =
+        sessionId != null && getIsTransactionPending(transactionBatch.status);
+      return isPending;
+    }
+  );
+
+  useEffect(() => {
+    if (pendingBatches.length > 0) {
```
Comment (StanislavSava) [inline src/components/TransactionsTracker/useTransactionsTracker.ts:30]: This check seems redundant, this is checked inside the checkTransactionStatus

File: src/hooks/transactions/useCheckTransactionStatus/checkBatch.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+}: TransactionStatusTrackerPropsType) {
+  const isBatchPending = sessionId != null && getIsTransactionPending(status);
+  try {
+    if (!isBatchPending || transactions == null) {
+      return;
+    }
+
+    const pendingTransactions = getPendingTransactions(transactions, timeouts);
+
+    if (pendingTransactions?.length === 0) {
```
Comment (StanislavSava) [inline src/hooks/transactions/useCheckTransactionStatus/checkBatch.ts:132]: if the batch is pending, it has pending transactions. Once all transactions go out of pending, only then the batch status changes

Comment (arhtudormorar) [reply src/hooks/transactions/useCheckTransactionStatus/checkBatch.ts]: ✅

**Pull request closed without merging**